### ISSUE TITLE
linux remote filesystem access fails

### ISSUE THREAD

**Issue by proxygoat** [glimpse-editor/Glimpse#321]
glimpse is unable to access remote filesystems in linux if the mount point is in the system /mnt directory

1. Start glimpse in a terminal
2. Open an image
3. Click on 'Export as…'
4. Select "Filesystem'
5. Select 'mnt'
6. See error dialog "Could not read the contents of mnt 
Error opening directory '/mnt': Permission denied"

 - OS: Linux
 - Version Ubuntu 18.04

See error message in terminal "(glimpse:nnnnn): GLib-GIO-WARNING **: 07:15:32.806: Error creating IO channel for /proc/self/mountinfo: Permission denied (g-file-error-quark, 2)"

Other applications are able to access mount points in /mnt without difficulty and without modifying the default permissions on /mnt itself.

**Comment by TrechNex** [comment]
Thanks for the report @proxygoat. Could you let us know how you installed Glimpse? That will help us determine who needs to look into this, as it seems to be a permissions issue.

**Comment by proxygoat** [comment]
Snap app, but I don't think that's germane. The default permissions on /mnt are rwxr-xr-x: anybody can read /mnt.

**Comment by TrechNex** [comment]
Tagging @hellsworth as they maintain snapcraft packaging and are best placed to help.

For background, Flatpaks and Snaps are "containerized" packages that are isolated from the wider system for security & stability reasons. As a result you have to grant them permissions to read from directories and mount points outside of your `/home` directory.

**Comment by proxygoat** [comment]
Didn't know that. Thanks.

**Comment by hellsworth** [comment]
@proxygoat sorry for the late reply since I've been traveling.. Anyways, @TrechNex is correct. The snap is built with access to files in the user's $HOME. One of the benefits of a containerized app is that it doesn't have access to anything on the system. Sometimes this can be inconvenient but if you just move your image to somewhere in $HOME, the glimpse-editor snap should work fine.

**Comment by hellsworth** [comment]
Actually, I just realized we are including the [removable-media plug](https://snapcraft.io/docs/removable-media-interface) that should allow for this. It is however not automatically connected. You can see at what plugs are connected with: `snap connections glimpse-editor`

So if you wish to still use files from /mnt, you'll need to connect to the interface with:
`sudo snap connect glimpse-editor:removable-media`

**Comment by proxygoat** [comment]
Major inconvenience in our situation, all of our images are
      resident on a server. Please consider also packaging as a .deb
      without this restriction.
    
    On 3/11/20 7:59 AM, Heather Ellsworth
      wrote:
    
    
      
      @proxygoat
        sorry for the late reply since I've been traveling.. Anyways, @TrechNex
        is correct. The snap is built with access to files in the user's
        $HOME. One of the benefits of a containerized app is that it
        doesn't have access to anything on the system. Sometimes this
        can be inconvenient but if you just move your image to somewhere
        in $HOME, the glimpse-editor snap should work fine.

**Comment by TrechNex** [comment]
For background, there is somewhere approaching a dozen different ways of packaging software for Linux. We are unable to support all of them!

This is because our project does not have unlimited resources or expertise, and we rely on the wider Linux community volunteering to package the software for us. In 0.1.2 we created source tarballs specifically for distribution packagers to use, and in 0.1.4 we plan to refactor "libgimp" to "libglimpse" so there are no conflicts if you install Glimpse Image Editor at the same time.

The snapcraft packaging is something @hellsworth and @psymole have volunteered to do on our behalf, and we are deeply grateful for what they have contributed :)

As a project the format we settled on was Flatpak (on Flathub), and a fallback installation method (AppImage) is something we're actively working on. The reason we chose those formats specifically is that they strike the right balance of compatibility and maintainability for us.

If you would like a Debian package, I recommend you submit a request to the Debian or Ubuntu projects as one of their maintainers may be able to pick it up.
